src/snowcli/cli/streamlit/commands.py

```diff
@@ -169,6 +166,12 @@ def streamlit_deploy(
     open_: bool = typer.Option(
         False, "--open", help="Whether to open Streamlit in a browser.", is_flag=True
     ),
+    experimental: bool = typer.Option(
+        False,
+        "--experimental",
+        help="Use experimental new embedded stage behavior",
+        is_flag=True,
+    ),
```

Comment:
@sfc-gh-pjob created decorator for experimental flag, here is the commit https://github.com/Snowflake-Labs/snowcli/commit/e43eff364e7c8c3932e59ca365e65d789e234625.

You can use `@with_experimental_behaviour()` on command

Reply:
Sweet -- I'll refactor to use that!

Reply:
@sfc-gh-pjob I tried using that, but for some reason the flag doesn't seem to be getting passed through, either in the pytest function or when I test it locally. Any suggestions for what I might have done wrong?

Reply:
Fixed!